lib/matplotlib/collections.py

```diff
@@ -1396,30 +1437,31 @@ def __init__(self, segments,     # Can be None.
            "interior" can be specified by appropriate usage of
            `~.path.Path.CLOSEPOLY`.
         **kwargs
-            Forwareded to `.Collection`.
+            Forwarded to `.Collection`.
         """
-        if colors is None:
-            colors = mpl.rcParams['lines.color']
-        if linewidths is None:
-            linewidths = (mpl.rcParams['lines.linewidth'],)
-        if antialiaseds is None:
-            antialiaseds = (mpl.rcParams['lines.antialiased'],)
+        kw_plural = dict(linewidths=linewidths,
+                         colors=colors,
+                         facecolors=facecolors,
+                         antialiaseds=antialiaseds,
+                         linestyles=linestyles,)
+
+        kw = {k: kwargs.pop(k[:-1], val) for k, val in kw_plural.items()}
+        kw.update(kwargs)
+        colors = kw.pop('colors')
```

Comment:
Lines 1442 to 1450 should  should be grouped together.
```suggestion
        kw_plural = dict(linewidths=linewidths,
                         colors=colors,
                         facecolors=facecolors,
                         antialiaseds=antialiaseds,
                         linestyles=linestyles,)
        kw = {k: kwargs.pop(k[:-1], val) for k, val in kw_plural.items()}
        kw.update(kwargs)
        colors = kw.pop('colors')

```
(diff looks more complicated than it is, just a moved newline)
This should get a comment as it's non-trivial to follow what it's doing (handling singular/plural kwargs, precedence, transforming the explicit keywords into the kw-dict (which also implies, they should not be used anymore hereafter).

Alternatively, it's better to keep this explicit. Isn't this the same as doing explicitly:

~~~
linewidths = kw.pop('linewidth', linewidths)
...
~~~
If not, still some `linewidths = some_helper_func('linewidth', linewidths)` should be possible.

Generally, this seems unrelated to the core of the PR. While it's not too complicated, I'd prefer it in a separate PR, because the PR is already herculean task.

Reply:
I've reworked things to eliminate these kwarg manipulations.  (Changeset will be pushed shortly.)